merkle-tree/concurrent/src/lib.rs

```diff
@@ -59,8 +59,7 @@ where
     pub roots_length: usize,
     pub current_root_index: usize,
 
-    // TODO(vadorovsky): Add canopy.
-    pub canopy_depth: usize,
+    pub canopy_depth: u32,
```

Comment:
even u8 would be enough right?
It's fighting over scraps just saying.

Reply:
I picked u32, because all `.pow()` implementations expect exponential to be u32, e.g.

https://doc.rust-lang.org/std/primitive.usize.html#method.pow

And we are using pow in `canopy_size` method (basically `2_usize.pow(canopy_depth - 1) - 1`) to figure out the number of nodes in canopy.